torch/csrc/jit/ir_import.cpp

```diff
@@ -0,0 +1,559 @@
+        case onnx_AttributeProto_AttributeType_UNDEFINED:
+          throw std::runtime_error("UNDEFINED attribute unsupported");
+          break;
+        case onnx_AttributeProto_AttributeType_FLOAT:
+          node->f_(name, attr.f);
+          break;
+        case onnx_AttributeProto_AttributeType_INT:
+          node->i_(name, attr.i);
+          break;
+        case onnx_AttributeProto_AttributeType_STRING:
+          node->s_(name, std::move(attr.s));
+          break;
+        case onnx_AttributeProto_AttributeType_TENSOR:
+          node->t_(name, buildTensor(attr.t));
+          break;
+        case onnx_AttributeProto_AttributeType_GRAPH:
+          node->g_(name, buildGraph(readSubgraph(attr.g)));
+          break;
+        case onnx_AttributeProto_AttributeType_FLOATS:
+          node->fs_(name, std::move(attr.fs));
+          break;
+        case onnx_AttributeProto_AttributeType_INTS:
+          node->is_(name, std::move(attr.is));
+          break;
+        case onnx_AttributeProto_AttributeType_STRINGS:
+          node->ss_(name, std::move(attr.ss));
+          break;
+        case onnx_AttributeProto_AttributeType_TENSORS:
+          node->ts_(name, buildTensors(attr.ts));
+          break;
+        case onnx_AttributeProto_AttributeType_GRAPHS:
+          node->gs_(name, buildGraphs(readSubgraphs(attr.gs)));
+          break;
+      }
+    }
+
+    for (auto input : node_.inputs) {
+      auto v = value_map[input];
+      node->addInput(v);
+    }
+
+    for (size_t i=0; i<node_.outputs.size(); i++) {
+      value_map[node_.outputs[i]] = node->outputs()[i];
+    }
+
+    block->appendNode(node);
+  }
+
+  for (auto output : graph_.outputs) {
+    if (value_map.find(output.name) != value_map.end()) {
+      Value* v = value_map.at(output.name);
+      block->registerOutput(v);
+    }
+  }
+}
+
+void buildBlocks(const std::vector<Graph_>& graphs_, Node* node,
+                 std::unordered_map<std::string, Value*>& value_map) {
+  for (auto g_ : graphs_) {
+    auto block = node->addBlock();
+    buildBlock(g_, block, value_map);
+  }
+}
+
+std::shared_ptr<Graph> buildGraph(const Graph_& graph_) {
+  auto graph = std::make_shared<Graph>();
+  std::unordered_map<std::string, Value*> value_map;
+
+  buildBlock(graph_, graph->block(), value_map);
+
+  return graph;
+}
+
+std::vector<std::shared_ptr<Graph>> buildGraphs(const std::vector<Graph_>& graphs_) {
+  std::vector<std::shared_ptr<Graph>> graphs;
+  for (auto g_ : graphs_) {
+    graphs.push_back(buildGraph(g_));
+  }
+  return graphs;
+}
+
+std::shared_ptr<Graph> buildGraph(const Graph_& graph_, std::vector<at::Tensor>& initializers) {
+
+  auto graph = buildGraph(graph_);
+
+  for (auto tensor_ : graph_.initializers) {
+    initializers.push_back(buildTensor(tensor_));
+  }
+
+  return graph;
+}
+
+}
+
+std::shared_ptr<Graph> ImportIRGraph(const std::string& serialized_graph,
+                                     std::vector<at::Tensor>& initializers) {
```

Comment:
Do we use initializers for anything? Are they an output argument?

Reply:
Do we use them for anything though? Are they ever non-empty?

Reply:
See comment below